sys/src/9/amd64/multiboot.c

```diff
@@ -164,17 +189,25 @@ multiboot(u32 magic, u32 pmbi, int vflag)
 			if(mod->string != 0)
 				p = KADDR(mod->string);
 			if(vflag)
-				print("mod %#x %#x <%s>\n",
+				print("Multiboot: mod %#x %#x <%s>\n",
 				      mod->modstart, mod->modend, p);
 			else {
+				u64 modstart = mod->modstart;
 				usize len = mod->modend - mod->modstart;
-				pamapinsert(mod->modstart, len, PamMODULE);
+				pamapinsert(modstart, len, PamMODULE);
```

Comment:
did you check to make sure the pam code adds PTEs for this region.

Reply:
I'm not sure it is.  PamMODULE is not handled in meminit().  When I add it I see
```
panic: cpu0: mmukphysmap: remapping kernel direct address at va 0xffff800012a01044 (pml1 0xffff80000280a000 old 0x8000000012a01001 new 0x8000000012a01047)
cpu0 stack: k:ffff80000029e052 k:ffff800000248a22 k:ffff800000248b95 k:ffff800000246fdb k:ffff800000246487 ?:0
```

One thing I notice is that the area is not rounded to 4k.  Not yet sure if that's the particular problem here though.